db/syncPostsToGrapher.ts

```diff
@@ -23,14 +32,46 @@ const syncPostsToGrapher = async (): Promise<void> => {
     const toDelete = existsInGrapher
         .filter((p) => !doesExistInWordpress[p.id])
         .map((p) => p.id)
+    const replacer = (
+        _match: string,
+        _firstPattern: string,
+        _offset: number,
+        fullString: string,
+        matches: Record<string, string>
+    ): string => {
+        const block = allBlocks[matches["id"].toString()]
+        if (block) return block.post_content
+        else return fullString
+    }
     const toInsert = rows.map((post: any) => {
+        // Now resolve references by using replace with a regex and a function.
+        // There are two special cases - one is that not all refs resolve (🤨)
+        // in which case we leave the original ref comment as is; the second is
+        // that some blocks reference other blocks (🎉 recursion!) and so we
+        // check in a loop if the regexp still matches and run replace again.
+        // Because not all refs resolve we have to limit the number of attempts - for now
```

Comment:
So as I understand it, if we weren't limiting the recursion, we would then go over the same unresolvable reference indefinitely. One way to solve this would be to remove that reference altogether, which leaves me wondering: what scenario do you anticipate in which it would be useful to have a reference to a reusable block that doesn't dereference?